sos/collector/transports/oc.py

```diff
@@ -0,0 +1,206 @@
+                        "tty": True
+                    }
+                ],
+                "restartPolicy": "Never",
+                "nodeName": self.address,
+                "hostNetwork": True,
+                "hostPID": True,
+                "hostIPC": True
+            }
+        }
+
+    def _connect(self, password):
+        # the oc binary must be _locally_ available for this to work
+        if not is_executable('oc'):
+            return False
+
+        # deploy the debug container we'll exec into
+        podconf = self.get_node_pod_config()
+        self.pod_name = podconf['metadata']['name']
+        fd, self.pod_tmp_conf = tempfile.mkstemp(dir=self.tmpdir)
+        with open(fd, 'w') as cfile:
+            json.dump(podconf, cfile)
+        self.log_debug("Starting sos collector container '%s'" % self.pod_name)
+        out = sos_get_command_output(
+            "oc create -f %s" % self.pod_tmp_conf
+        )
+        if (out['status'] != 0 or "pod/%s created" % self.pod_name not in
+                out['output']):
+            self.log_error("Unable to deploy sos collect pod")
+            self.log_debug("Debug pod deployment failed: %s" % out['output'])
+            return False
+        self.log_debug("Pod '%s' successfully deployed, waiting for pod to "
+                       "enter running state" % self.pod_name)
+
+        # wait for the pod to report as running
+        try:
+            up = sos_get_command_output("oc wait --for=condition=Ready pod/%s "
+                                        "--timeout=30s" % self.pod_name,
+                                        # timeout is for local safety, not oc
+                                        timeout=40)
+            if not up['status'] == 0:
+                self.log_error("Pod not available after 30 seconds")
+                return False
+        except SoSTimeoutError:
+            self.log_error("Timeout while polling for pod readiness")
+            return False
+        except Exception as err:
+            self.log_error("Error while waiting for pod to be ready: %s"
+                           % err)
+            return False
+
+        return True
+
+    def _format_cmd_for_exec(self, cmd):
+        if cmd.startswith('oc'):
+            return ("oc exec --request-timeout=0 %s -- chroot /host %s"
+                    % (self.pod_name, cmd))
+        return super(OCTransport, self)._format_cmd_for_exec(cmd)
+
+    def run_command(self, cmd, timeout=180, need_root=False, env=None,
+                    get_pty=False):
+        # debug pod setup is slow, extend all timeouts to account for this
+        if timeout:
+            timeout += 10
+
+        # since we always execute within a bash shell, force disable get_pty
+        # to avoid double-quoting
+        get_pty = False
+        return super(OCTransport, self).run_command(cmd, timeout, need_root,
+                                                    env, get_pty)
```

Comment:
I understand we need `get_pty` as method argument since the method can be called somewhere with the argument that we do ignore here. Isn't it easier to ignore `get_pty` variable, then, and cal `run_command(cmd, timeout, need_root, env, False)` directly?